src/webgpu/shader/validation/uniformity/uniformity.spec.ts

```diff
@@ -2693,3 +2811,30 @@ fn main() {
     }
     t.expectCompileResult(res, code);
   });
+
+g.test('subgroups,parameters')
+  .desc('Test subgroup operations that require a uniform parameter')
+  .params(u =>
+    u
+      .combine('op', ['subgroupShuffleUp', 'subgroupShuffleDown', 'subgroupShuffleXor'] as const)
+      .combine('uniform', [false, true] as const)
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase('subgroups' as GPUFeatureName);
+  })
+  .fn(t => {
+    const wgsl = `
+enable subgroups;
+
+var<private> non_uniform : u32 = 0;
+
+@group(0) @binding(0)
+var<storage> uniform : u32;
```

Comment:
the variable name is slightly confusing because it shadows the address space by the same name.

But I see why you use a variable at all; it's a tighter rule to check than a const.